modules/playlists/src/main/java/org/opencastproject/playlists/PlaylistService.java

```diff
@@ -423,10 +416,13 @@ public JaxbPlaylist enrich(Playlist playlist) {
       try {
         JaxbPlaylistEntry entry = jaxbPlaylistEntries.get(i);
         if (entry.getType() == PlaylistEntryType.EVENT) {
-          Opt<Event> optEvent = indexService.getEvent(jaxbPlaylistEntries.get(i).getContentId(), elasticsearchIndex);
-          // We only get an event from the indexService if we have permission to do so (and if it exists ofc)
-          if (optEvent.isSome()) {
-            Event event = optEvent.get();
+
+          // We only get an event from the index if we have permission to do so (and if it exists ofc)
```

Comment:
Is this actually true? We're talking directly to the index right? The index has the acl, but doesn't know what to do with it. It can't check permissions like this, you have to do that.

Reply:
As discussed in chat, the ES index does actually do permission checking, I think. Either way, the index service didn't do much more, just retrieve the event from the index.

Reply:
At least, the permissions are stored in the index. I'll take a look later.

Reply:
Permissions are checked.